### PR TITLE
Fix and improve linting / update requirements

### PR DESCRIPTION
**Pull request by adri0**
- Replace black and isort by ruff.
- Add makefile to run all linting using `make lint`.
- Add lint job to github build workflow.
- Rename test.yml to build.yml
- Fix linting issues.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -28,3 +28,22 @@ jobs:
       - name: Test with pytest
         run: |
           pytest
+
+  lint:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
```
Comment (mkopec87) [inline .github/workflows/build.yml:35]: pycharm says that fail-fast is only for matrix strategy

File: requirements.txt
Comment (mkopec87) [inline requirements.txt:1]: Where is this file used?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -28,3 +28,22 @@ jobs:
...
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python 3.11
+        uses: actions/setup-python@v3
+        with:
+          python-version: 3.11
+      - name: Install dependencies
+        run: python -m pip install .[dev,test]
```
Comment (mkopec87) [inline .github/workflows/build.yml:43]: is "test" scope needed?

**Review by mkopec87 — APPROVED** [review]
Looks good to me :)

Maybe worth removing the "strategy" from "lint" build step?

File: .github/workflows/build.yml (hunk 2 above)
Comment (adri0) [reply .github/workflows/build.yml:43]: Yes, because it lints the tests as well :\ So it's needed to infer references from pytest.

File: requirements.txt
Comment (adri0) [reply requirements.txt:1]: It's where I'm keeping the pinned dependencies. But you're right it's not being used. I just generate it once in a while from `pip freeze`.

**Pull request merged**